psydac/api/fem.py

```diff
@@ -994,7 +997,10 @@ def assemble(self, *, reset=True, **kwargs):
             self._vector.exchange_assembly_data()
 
         if self._vector: self._vector.ghost_regions_in_sync = False
-        return self._vector
+        if self._vector == None : 
+            return None
+        else : 
+            return self._vector.copy()
```

Comment:
Same comment as above, but the extra memory usage is less important in the case of a vector.